concrete-core/src/backends/core/implementation/engines/glwe_ciphertext_vector_trivial_encryption.rs

```diff
@@ -0,0 +1,169 @@
+{
+    /// # Example:
+    ///
+    /// ```
+    /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+    ///
+    /// use concrete_commons::dispersion::Variance;
+    /// use concrete_commons::parameters::{GlweDimension, PolynomialSize};
+    /// use concrete_core::prelude::*;
+    ///
+    /// // DISCLAIMER: the parameters used here are only for test purpose, and are not secure.
+    /// let glwe_dimension = GlweDimension(2);
+    /// let polynomial_size = PolynomialSize(4);
+    /// let input = vec![3_u32 << 20; 2 * polynomial_size.0];
+    /// let ciphertext_count = GlweCiphertextCount(2);
+    ///
+    /// let mut engine = CoreEngine::new()?;
+    /// let plaintext_vector: PlaintextVector32 = engine.create_plaintext_vector(&input)?;
+    /// let ciphertext_vector: GlweCiphertextVector32 = engine
+    ///     .trivially_encrypt_glwe_ciphertext_vector(
+    ///         glwe_dimension.to_glwe_size(),
+    ///         ciphertext_count,
+    ///         &plaintext_vector,
+    ///     )?;
+    ///
+    /// assert_eq!(ciphertext_vector.glwe_dimension(), glwe_dimension);
+    /// assert_eq!(ciphertext_vector.polynomial_size(), polynomial_size);
+    /// assert_eq!(ciphertext_vector.glwe_ciphertext_count(), ciphertext_count);
+    ///
+    /// engine.destroy(plaintext_vector)?;
+    /// engine.destroy(ciphertext_vector)?;
+    ///
+    /// # Ok(())
+    /// # }
+    /// ```
+    fn trivially_encrypt_glwe_ciphertext_vector(
+        &mut self,
+        glwe_size: GlweSize,
+        glwe_ciphertext_count: GlweCiphertextCount,
+        input: &PlaintextVector32,
+    ) -> Result<GlweCiphertextVector32, GlweCiphertextVectorTrivialEncryptionError<Self::EngineError>>
+    {
+        unsafe {
+            Ok(self.trivially_encrypt_glwe_ciphertext_vector_unchecked(
+                glwe_size,
+                glwe_ciphertext_count,
+                input,
+            ))
+        }
+    }
+
+    unsafe fn trivially_encrypt_glwe_ciphertext_vector_unchecked(
+        &mut self,
+        glwe_size: GlweSize,
+        glwe_ciphertext_count: GlweCiphertextCount,
+        input: &PlaintextVector32,
+    ) -> GlweCiphertextVector32 {
+        let mut vec = Vec::with_capacity(glwe_ciphertext_count.0 * glwe_size.0);
+        let input_slice = input.0.as_tensor().as_slice();
+        let poly_size = PolynomialSize(input_slice.chunks(glwe_ciphertext_count.0).count());
+        for chunk in input_slice.chunks(glwe_ciphertext_count.0) {
+            let ciphertext =
+                ImplGlwe::new_trivial_encryption(glwe_size, &PlaintextList::from_container(chunk));
+            for p in ciphertext.into_polynomial_list().polynomial_iter() {
+                for c in p.coefficient_iter() {
+                    vec.push(*c);
+                }
+            }
+        }
+        let ciphertext_vector: ImplGlweList<Vec<u32>> =
+            ImplGlweList::from_container(vec, glwe_size.to_glwe_dimension(), poly_size);
+        GlweCiphertextVector32(ciphertext_vector)
```

Comment:
Here there are a lot of things which already exist in the codebase. It would be better to use: 
+ `GlweList::allocate`
+ `GlweList::ciphertext_iter_mut`
+ `GlweCiphertext::get_body`
+ `Tensor::fill_with_copy`

Actually, it would be even better to implement a private entry point `GlweCiphertext::fill_with_trivial_encrypt` and then to use it.

Reply:
I just modified the commit to follow this recommendation.